cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tether/TetherUtils.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.tether;
+
+import com.google.common.net.HttpHeaders;
+import io.cdap.cdap.common.internal.remote.RemoteAuthenticator;
+import io.cdap.common.http.HttpMethod;
+import io.cdap.common.http.HttpRequestConfig;
+import io.cdap.common.http.HttpRequests;
+import io.cdap.common.http.HttpResponse;
+
+import java.io.IOException;
+import java.net.URI;
+import javax.annotation.Nullable;
+
+/**
+ * Utility functions for tethering.
+ */
+public class TetherUtils {
+  // Connection timeout = 5 seconds.
+  private static final int TIMEOUT_MS = 5000;
+  private static final RemoteAuthenticator AUTHENTICATOR = RemoteAuthenticator.getDefaultAuthenticator();
```

Comment:
Do we need to use something similar to RuntimeMonitor?
https://github.com/cdapio/cdap/blob/develop/cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/monitor/RuntimeMonitors.java#L113

Reply:
hm why is that needed? can't we just use `cdap.remote.authenticator.class`? https://github.com/cdapio/cdap/blob/develop/cdap-common/src/main/java/io/cdap/cdap/common/internal/remote/RemoteAuthenticator.java#L36

Reply:
Now calling RemoteAuthenticator.setDefaultAuthenticator() in RemoteAgentService.startUp()